### PR TITLE
impl RecordBatchStream for QueryResults

### PR DESCRIPTION
**Pull request by alamb**
While trying out https://github.com/apache/arrow-datafusion/pull/2226 in IOx, I found it would be super convenient to have `QueryResults` be a `RecordBatchStream` (so it has the same interface as the current executions)

### CODE REVIEW

File: datafusion/scheduler/src/task.rs [hunk 1]
```diff
@@ -203,15 +209,18 @@ impl Task {
...
+    schema: SchemaRef,
 
     /// Keep a reference to the [`QueryTask`] so it isn't dropped early
     #[allow(unused)]
     query: Arc<QueryTask>,
 }
 
 impl Stream for QueryResults {
-    type Item = Result<RecordBatch>;
+    type Item = ArrowResult<RecordBatch>;
```
Comment (tustvold) [inline datafusion/scheduler/src/task.rs:223]: This is one quirk I was hoping to take the opportunity to fix, that DataFusion streams don't return a DataFusion error... It isn't the end of the world, but I'm curious how important this aspect is?

Comment (tustvold) [inline datafusion/scheduler/src/task.rs:223]: This is one quirk I was hoping to take the opportunity to fix, that DataFusion streams don't return a DataFusion error... It isn't the end of the world, but I'm curious how important this aspect is?

Comment (alamb) [reply datafusion/scheduler/src/task.rs:223]: I think it is unimportant in the overall design sense

I think it is critically important from the "minimize existing code" sense

**Comment by tustvold** [comment]
Implemented as part of https://github.com/apache/arrow-datafusion/pull/2226/commits/505e880ad5d3a3e1714c98942ce207959ab3f2d1

**Pull request closed without merging**